app/controllers/application_controller.rb

```diff
@@ -34,6 +35,13 @@ def confirm_admin_user_async
   end
 
   def redirect_unless_has_data_access
-    redirect_to root_url unless current_user.allowed_data_access?
+    redirect_to root_path unless current_user.allowed_data_access?
+  end
+
+  def confirm_user_not_disabled!
+    if current_user && current_user.disabled_by_fund?
```

Comment:
Currently this treats `current_user == false` the same as and enabled user, which seems like a bad plan. I don't think you can get here with current_user being false (authenticate would have kicked you out of the before_action stack and canceled the rest of em), but it might be good to clean that up.

Reply:
Took out the first current user. This should now choke if current user isn't set, which is good news I think!